extensions/stats/config.h

```diff
@@ -0,0 +1,384 @@
+          auto val = (expr);                                             \
+          logDebug(absl::StrCat((key), "=", ToString(val)));             \
+          return (SYMIFEMPTY(ToString(val), (sym)));                     \
+        }                                                                \
+  }
+#define MAPPING(key, expr) MAPPING_SYM((key), (expr), unknown)
+
+// Example Prometheus output
+//
+// istio_requests_total{
+// connection_security_policy="unknown",
+// destination_app="svc01-0-8",
+// destination_principal="unknown",
+// destination_service="svc01-0-8.service-graph01.svc.cluster.local",
+// destination_service_name="svc01-0-8",
+// destination_service_namespace="service-graph01",
+// destination_version="v1",
+// destination_workload="svc01-0-8",
+// destination_workload_namespace="service-graph01",
+// permissive_response_code="none",
+// permissive_response_policyid="none",
+// reporter="source",
+// request_protocol="http",
+// response_code="200",
+// response_flags="-",
+// source_app="svc01-0",
+// source_principal="unknown",
+// source_version="v2",
+// source_workload="svc01-0v2",
+// source_workload_namespace="service-graph01"
+// }
+
+std::vector<Mapping> getStandardMappings() {
+  return {
+      MAPPING("reporter", (outbound ? vSource : vDest)),
+      // --> Peer info source
+      MAPPING("source_workload", source.workload_name()),
+      MAPPING("source_workload_namespace", source.namespace_()),
+      MAPPING("source_principal", requestInfo.source_principal),
+      MAPPING("source_app", source_labels["app"]),
+      MAPPING("source_version", source_labels["version"]),
+
+      // --> Peer info destination
+      MAPPING("destination_workload", dest.workload_name()),
+      MAPPING("destination_workload_namespace", dest.namespace_()),
+      MAPPING("destination_principal", requestInfo.destination_principal),
+      MAPPING("destination_app", dest_labels["app"]),
+      MAPPING("destination_version", dest_labels["version"]),
+
+      // --> Service info
+      MAPPING("destination_service_host", requestInfo.destination_service_host),
+      MAPPING("destination_service_name", dest.workload_name()),
+      MAPPING("destination_service_namespace", dest.namespace_()),
+
+      MAPPING("request_protocol", requestInfo.request_protocol),
+      MAPPING("response_code", requestInfo.response_code),
+      MAPPING_SYM("response_flags", requestInfo.response_flag, vDash),
+
+      MAPPING("connection_security_policy",
+              (outbound ? unknown : (requestInfo.mTLS ? vMTLS : vNone))),
+      MAPPING_SYM("permissive_response_code",
+                  requestInfo.permissive_response_code, vNone),
+      MAPPING_SYM("permissive_response_policyid",
+                  requestInfo.permissive_response_policyid, vNone)};
+}
+
+// StatGen is dimensioned using standard Istio dimensions.
+// Standard Istio metrics have the following dimensions
+class StatGen {
```

Comment:
It's not clear to me whether `StatGen` is supposed to be copyable or moveable, or shared. I cannot image

Some indicator could be
declare fields as const
explicitly declare copy constructor/move constructor as default.